src/utils/components/SSHSecretSection/utils/components/SelectSecret.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC, ReactElement, useState } from 'react';
+
+import { SecretModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getName } from '@kubevirt-utils/resources/shared';
+import { validateSSHPublicKey } from '@kubevirt-utils/utils/utils';
+import {
+  getGroupVersionKindForModel,
+  ResourceLink,
+  WatchK8sResult,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { Alert, AlertVariant, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import Loading from '../../../Loading/Loading';
+import { decodeSecret } from '../utils';
+
+type SelectSecretProps = {
+  secretsResourceData: WatchK8sResult<IoK8sApiCoreV1Secret[]>;
+  selectedSecretName: string;
+  onSelectSecret: (secretName: string) => void;
+  id?: string;
+};
+
+const SelectSecret: FC<SelectSecretProps> = ({
+  secretsResourceData,
+  selectedSecretName,
+  onSelectSecret,
+  id,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isSecretSelectOpen, setSecretSelectOpen] = useState(false);
+
+  const [secrets, secretsLoaded, secretsError] = secretsResourceData;
+  const sshKeySecrets = secrets
+    ? secrets?.filter((secret) => validateSSHPublicKey(decodeSecret(secret)))
+    : [];
+
+  const filterSecrets = (_, userInput: string): ReactElement[] =>
+    sshKeySecrets
+      ?.filter((secret) => getName(secret)?.includes(userInput))
+      ?.map((secret) => (
+        <SelectOption key={getName(secret)} value={getName(secret)}>
+          <ResourceLink
+            groupVersionKind={getGroupVersionKindForModel(SecretModel)}
+            name={getName(secret)}
+          />
+        </SelectOption>
+      )) as ReactElement[];
+
+  const onSelect = (event, newValue) => {
+    onSelectSecret(newValue);
+    setSecretSelectOpen(false);
+  };
+
+  if (!secretsLoaded) return <Loading />;
+
+  if (secretsError)
+    return (
+      <Alert title={t('Error')} isInline variant={AlertVariant.danger}>
+        {secretsError?.message}
+      </Alert>
+    );
+
+  return (
+    <Select
+      menuAppendTo="parent"
+      isOpen={isSecretSelectOpen}
+      onToggle={() => setSecretSelectOpen(!isSecretSelectOpen)}
+      onSelect={onSelect}
+      variant={SelectVariant.single}
+      onFilter={filterSecrets}
+      hasInlineFilter
+      selections={selectedSecretName}
+      placeholderText={t('--- Select secret ---')}
+      maxHeight={400}
+      id={id || 'select-secret'}
+    >
+      {sshKeySecrets?.map((secret) => {
+        const secretName = getName(secret);
+        return (
+          <SelectOption key={secretName} value={secretName}>
+            <ResourceLink
+              groupVersionKind={getGroupVersionKindForModel(SecretModel)}
+              name={secretName}
+            />
+          </SelectOption>
+        );
+      })}
```

Comment:
this block and lines 42-49 look very similar, does it make sense to create a function for this? or a component?

Reply:
That's a good suggestion. Done!